src/Microsoft.OData.Core/JsonLight/ODataJsonLightPropertySerializer.cs

```diff
@@ -389,6 +396,30 @@ private void WriteStreamReferenceProperty(string propertyName, ODataStreamRefere
             }
         }
 
+        /// <summary>
+        /// Writes a resource property.
+        /// </summary>
+        /// <param name="property">The property to write out.</param>
+        /// <param name="resourceValue">The resource value to be written</param>
+        /// <param name="isOpenPropertyType">If the property is open.</param>
+        private void WriteResourceProperty(
+            ODataProperty property,
+            ODataResourceValue resourceValue,
+            bool isOpenPropertyType)
+        {
+            if (!this.currentPropertyInfo.IsTopLevel)

```

Comment:
would an ODataResourceValue ever be toplevel?

Reply:
We don't limit end user to create a property with value as ODataResourceValue. 
In my opinion, We should not limit end user to write a top level property with value as "ODataResourceValue".

Reply:
Now that we prohibit writing a resource value at the top level, can we get rid of this code that checks to see if it's top level?

---
In reply to: [241587425](https://github.com/OData/odata.net/pull/1346#discussion_r241587425) [](ancestors = 241587425)

Reply:
@mikepizzo removed. #Resolved.